rfcs/react-components/convergence/theme-shared-colors.md

```diff
@@ -0,0 +1,114 @@
+4. tried to remove even more tokens.
+
+As measured in Chrome profiler, there has been a significant difference in reflow times (see the RFC for details).
+
+Flamegrill showed us following numbers in the synthetic test:
+
+| Theme structure               |  ms |
+| ----------------------------- | --: |
+| 1200 tokens, alias references | 490 |
+| 1200 tokens, inline values    | 410 |
+| 550 tokens                    | 250 |
+| 160 tokens                    | 150 |
+
+The test showed that there was a clear performance hit depending on the number of the CSS variables. As a result:
+
+1. we got rid of alias references and inlined the values,
+2. removed global color tokens from the theme,
+3. flattened the theme object.
+
+This reduced the tokens from 1200 to 550 and improved the perf from 490ms (1st row in the table) to 250ms (3rd row).
+
+Since then, we have added couple of tokens as requested by design, as of today the theme object contains ~650 tokens.
+
+### Real Application Performance
+
+To avoid any performance regressions there are performance gates running as part of partner application CI.
+
+The test case which is interesting for us is WLT hot - that opens a new application window (pre-heat), closes the window and opens it again measuring the performance. The current time for this test is around 700ms.
+
+Just adding v9 `Provider` with a theme, the test time rises by **20ms** (with the original approach having 1200 tokens and alias references it was 80ms) which does not pass the gate. For that reason we have been stuck on this for 3 months already not being able to enable the v9 `Provider` in the application, blocking feature teams in starting migration to v9.
+
+As an experiment we tried to reduce the tokens to 200, the test time degradation was 7ms.
+
+| Theme structure               | WLT degradation [ms] |
+| ----------------------------- | -------------------: |
+| 1200 tokens, alias references |                   70 |
+| 600 tokens                    |                   20 |
+| 200 tokens                    |                    7 |
+
+**The numbers confirm the performance decreases linearly depending on the number of CSS variables (and therefore theme tokens).**
+
+### New Shared Color Alias tokens request
+
+_The numbers might be inaccurate as the feature is not fully spec-ed yet!_
+
+In order to support new scenarios (colorful calendar view), there is a new request to add 3 new shared color alias tokens. Adding them would result in adding 3 alias tokens for each of 49 shared colors = adding 147 new shared color alias tokens which would affect application performance.
+
+## Detailed Design or Proposal
+
+The theme currently contains 49 global shared colors.
+
+Current alias structure forces each ramp to be uniform (the same set of alias tokens for each shared color), resulting in 49 alias tokens each time we need a new line item (new alias token). Today we have 9 alias slots = total alias count is 441 (49x9).
```

Comment:
**q:** can you pls elaborate what `ramp` means?

Reply:
the same set of alias tokens for each shared color :-) I will try to describe it better.